// semantic-ui-calendar-react/dist/views/DayView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.WEEKS_TO_DISPLAY = exports.DAY_CALENDAR_ROW_WIDTH = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Calendar = _interopRequireDefault(require("./Calendar"));

var _Header = _interopRequireDefault(require("./CalendarHeader/Header"));

var _Body = _interopRequireDefault(require("./CalendarBody/Body"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var DAY_CALENDAR_ROW_WIDTH = '7';
exports.DAY_CALENDAR_ROW_WIDTH = DAY_CALENDAR_ROW_WIDTH;
var WEEKS_TO_DISPLAY = 6;
exports.WEEKS_TO_DISPLAY = WEEKS_TO_DISPLAY;

function DayView(props) {
  var days = props.days,
      onNextPageBtnClick = props.onNextPageBtnClick,
      onPrevPageBtnClick = props.onPrevPageBtnClick,
      onDayClick = props.onDayClick,
      hasNextPage = props.hasNextPage,
      hasPrevPage = props.hasPrevPage,
      currentDate = props.currentDate,
      onHeaderClick = props.onHeaderClick,
      disabled = props.disabled,
      active = props.active;
  return _react.default.createElement(_Calendar.default, null, _react.default.createElement(_Header.default, {
    width: DAY_CALENDAR_ROW_WIDTH,
    displayWeeks: true,
    onNextPageBtnClick: onNextPageBtnClick,
    onPrevPageBtnClick: onPrevPageBtnClick,
    hasNextPage: hasNextPage,
    hasPrevPage: hasPrevPage,
    title: currentDate,
    onHeaderClick: onHeaderClick
  }), _react.default.createElement(_Body.default, {
    width: DAY_CALENDAR_ROW_WIDTH,
    data: days,
    onCellClick: onDayClick,
    active: active,
    disabled: disabled
  }));
}

DayView.handledProps = ["active", "currentDate", "days", "disabled", "hasNextPage", "hasPrevPage", "onDayClick", "onHeaderClick", "onNextPageBtnClick", "onPrevPageBtnClick"];
DayView.propTypes = {
  /** An array of dates to fill a calendar with. */
  days: _propTypes.default.arrayOf(_propTypes.default.string).isRequired,

  /** Called after click on next page button. */
  onNextPageBtnClick: _propTypes.default.func.isRequired,

  /** Called after click on previous page button. */
  onPrevPageBtnClick: _propTypes.default.func.isRequired,

  /** Called after click on day. */
  onDayClick: _propTypes.default.func.isRequired,

  /** Whether to display previous page button as active or disabled. */
  hasPrevPage: _propTypes.default.bool.isRequired,

  /** Whether to display next page button as active or disabled. */
  hasNextPage: _propTypes.default.bool.isRequired,

  /** A date that is displayed in calendar header. */
  currentDate: _propTypes.default.string.isRequired,

  /** Called after click on calendar header. */
  onHeaderClick: _propTypes.default.func,

  /** An array of day positions to display as disabled. */
  disabled: _propTypes.default.arrayOf(_propTypes.default.number),

  /** Position of a day to display as active. */
  active: _propTypes.default.number
};
var _default = DayView;
exports.default = _default;
