// semantic-ui-calendar-react/dist/views/MinuteView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Calendar = _interopRequireDefault(require("./Calendar"));

var _Header = _interopRequireDefault(require("./CalendarHeader/Header"));

var _Body = _interopRequireDefault(require("./CalendarBody/Body"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MINUTE_CALENDAR_ROW_WIDTH = '3';

function MinuteView(props) {
  var minutes = props.minutes,
      hasHeader = props.hasHeader,
      onMinuteClick = props.onMinuteClick,
      onNextPageBtnClick = props.onNextPageBtnClick,
      onPrevPageBtnClick = props.onPrevPageBtnClick,
      hasNextPage = props.hasNextPage,
      hasPrevPage = props.hasPrevPage,
      onHeaderClick = props.onHeaderClick,
      active = props.active,
      currentDate = props.currentDate;
  var headerProps = {
    onHeaderClick: onHeaderClick,
    onNextPageBtnClick: onNextPageBtnClick,
    onPrevPageBtnClick: onPrevPageBtnClick,
    hasNextPage: hasNextPage,
    hasPrevPage: hasPrevPage,
    title: currentDate,
    width: MINUTE_CALENDAR_ROW_WIDTH,
    displayWeeks: false
  };
  return _react.default.createElement(_Calendar.default, null, hasHeader && _react.default.createElement(_Header.default, headerProps), _react.default.createElement(_Body.default, {
    width: MINUTE_CALENDAR_ROW_WIDTH,
    data: minutes,
    onCellClick: onMinuteClick,
    active: active
  }));
}

MinuteView.handledProps = ["active", "currentDate", "hasHeader", "hasNextPage", "hasPrevPage", "minutes", "onHeaderClick", "onMinuteClick", "onNextPageBtnClick", "onPrevPageBtnClick"];
MinuteView.propTypes = {
  /** Array of minutes to fill a calendar with. */
  minutes: _propTypes.default.arrayOf(_propTypes.default.string).isRequired,

  /** Wether to display header or not. */
  hasHeader: _propTypes.default.bool.isRequired,

  /** Called after click on minute. */
  onMinuteClick: _propTypes.default.func.isRequired,

  /** Called after click on next page button. */
  onNextPageBtnClick: _propTypes.default.func,

  /** Called after click on previous page button. */
  onPrevPageBtnClick: _propTypes.default.func,

  /** Whether to display previous page button as active or disabled. */
  hasPrevPage: _propTypes.default.bool,

  /** Whether to display next page button as active or disabled. */
  hasNextPage: _propTypes.default.bool,

  /** Called after click on calendar header. */
  onHeaderClick: _propTypes.default.func,

  /** Minute index to display as active. */
  active: _propTypes.default.number,

  /** A date that is displayed in calendar header. */
  currentDate: _propTypes.default.string
};
var _default = MinuteView;
exports.default = _default;
