// semantic-ui-calendar-react/dist/views/MonthView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _Calendar = _interopRequireDefault(require("./Calendar"));

var _Header = _interopRequireDefault(require("./CalendarHeader/Header"));

var _Body = _interopRequireDefault(require("./CalendarBody/Body"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var MONTH_CALENDAR_ROW_WIDTH = '3';

function MonthView(props) {
  var months = props.months,
      hasHeader = props.hasHeader,
      onMonthClick = props.onMonthClick,
      onNextPageBtnClick = props.onNextPageBtnClick,
      onPrevPageBtnClick = props.onPrevPageBtnClick,
      hasPrevPage = props.hasPrevPage,
      hasNextPage = props.hasNextPage,
      onHeaderClick = props.onHeaderClick,
      disabled = props.disabled,
      active = props.active,
      currentYear = props.currentYear;
  var headerProps = {
    onNextPageBtnClick: onNextPageBtnClick,
    onPrevPageBtnClick: onPrevPageBtnClick,
    hasPrevPage: hasPrevPage,
    hasNextPage: hasNextPage,
    onHeaderClick: onHeaderClick,
    title: currentYear,
    displayWeeks: false,
    width: MONTH_CALENDAR_ROW_WIDTH
  };
  return _react.default.createElement(_Calendar.default, null, hasHeader && _react.default.createElement(_Header.default, headerProps), _react.default.createElement(_Body.default, {
    width: MONTH_CALENDAR_ROW_WIDTH,
    data: months,
    onCellClick: onMonthClick,
    active: active,
    disabled: disabled
  }));
}

MonthView.handledProps = ["active", "currentYear", "disabled", "hasHeader", "hasNextPage", "hasPrevPage", "months", "onHeaderClick", "onMonthClick", "onNextPageBtnClick", "onPrevPageBtnClick"];
MonthView.propTypes = {
  /** Array of months to fill a calendar with. */
  months: _propTypes.default.arrayOf(_propTypes.default.string).isRequired,

  /** Wether to display header or not. */
  hasHeader: _propTypes.default.bool.isRequired,

  /** Called after click on month. */
  onMonthClick: _propTypes.default.func.isRequired,

  /** Called after click on next page button. */
  onNextPageBtnClick: _propTypes.default.func,

  /** Called after click on previous page button. */
  onPrevPageBtnClick: _propTypes.default.func,

  /** Whether to display previous page button as active or disabled. */
  hasPrevPage: _propTypes.default.bool,

  /** Whether to display next page button as active or disabled. */
  hasNextPage: _propTypes.default.bool,

  /** Called after click on calendar header. */
  onHeaderClick: _propTypes.default.func,

  /** An array of month indexes to display as disabled. */
  disabled: _propTypes.default.arrayOf(_propTypes.default.number),

  /** Index of a month that should be displayed as active. */
  active: _propTypes.default.number,

  /** A year to display in header. */
  currentYear: _propTypes.default.string
};
var _default = MonthView;
exports.default = _default;
