// semantic-ui-calendar-react/dist/views/YearView.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _lodash = _interopRequireDefault(require("lodash"));

var _Calendar = _interopRequireDefault(require("./Calendar"));

var _Header = _interopRequireDefault(require("./CalendarHeader/Header"));

var _Body = _interopRequireDefault(require("./CalendarBody/Body"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var YEAR_CALENDAR_ROW_WIDTH = '3';

function YearView(props) {
  var years = props.years,
      onNextPageBtnClick = props.onNextPageBtnClick,
      onPrevPageBtnClick = props.onPrevPageBtnClick,
      onYearClick = props.onYearClick,
      hasNextPage = props.hasNextPage,
      hasPrevPage = props.hasPrevPage,
      onHeaderClick = props.onHeaderClick,
      disabled = props.disabled,
      active = props.active;
  var headerTitle = "".concat(_lodash.default.first(years), " - ").concat(_lodash.default.last(years));
  return _react.default.createElement(_Calendar.default, null, _react.default.createElement(_Header.default, {
    title: headerTitle,
    onNextPageBtnClick: onNextPageBtnClick,
    onPrevPageBtnClick: onPrevPageBtnClick,
    hasNextPage: hasNextPage,
    hasPrevPage: hasPrevPage,
    onHeaderClick: onHeaderClick,
    width: YEAR_CALENDAR_ROW_WIDTH,
    displayWeeks: false
  }), _react.default.createElement(_Body.default, {
    width: YEAR_CALENDAR_ROW_WIDTH,
    data: years,
    onCellClick: onYearClick,
    active: active,
    disabled: disabled
  }));
}

YearView.handledProps = ["active", "disabled", "hasNextPage", "hasPrevPage", "onHeaderClick", "onNextPageBtnClick", "onPrevPageBtnClick", "onYearClick", "years"];
YearView.propTypes = {
  /** An array of years to fill a calendar with. */
  years: _propTypes.default.arrayOf(_propTypes.default.string).isRequired,

  /** Called after click on next page button. */
  onNextPageBtnClick: _propTypes.default.func.isRequired,

  /** Called after click on previous page button. */
  onPrevPageBtnClick: _propTypes.default.func.isRequired,

  /** Called after click on year. */
  onYearClick: _propTypes.default.func.isRequired,

  /** Whether to display previous page button as active or disabled. */
  hasPrevPage: _propTypes.default.bool.isRequired,

  /** Whether to display next page button as active or disabled. */
  hasNextPage: _propTypes.default.bool.isRequired,

  /** Called after click on calendar header. */
  onHeaderClick: _propTypes.default.func,

  /** An array of numbers that represent indexes of years in `years` array that should be displayed as disabled. */
  disabled: _propTypes.default.arrayOf(_propTypes.default.number),

  /** Index of a year in `years` array that should be displayed as active. */
  active: _propTypes.default.number
};
var _default = YearView;
exports.default = _default;
